src/main/java/com/allenai/ml/linalg/SparseVector.java

```diff
@@ -87,5 +89,5 @@ public double dotProduct(Vector other) {
           result += val * other.at(idx);
```

Comment:
@jakemannix I know in practice for vector space methods you run into sparse on sparse dot products. If I cared about `O(n)` vs `O(n lg n)` I would probably need to figure something algorithmically where I projected to a dense space. I'm not sure there's a clear superior solution across the board. What are your thoughts?